src/server/system_services/account_server.js

```diff
@@ -299,6 +299,48 @@ async function generate_account_keys(req) {
     });
 }
 
+/**
+ *
+ * UPDATE_ACCOUNT_KEYS
+ *
+ */
+async function update_account_keys(req) {
+    const account = system_store.get_account_by_email(req.rpc_params.email);
+    if (!account) {
+        throw new RpcError('NO_SUCH_ACCOUNT', 'No such account email: ' + req.rpc_params.email);
+    }
+    if (req.system && req.account) {
+        if (!is_support_or_admin_or_me(req.system, req.account, account)) {
```

Comment:
The "req.system && req.account" condition is only to verify that these values are defined?
So you can continue the same condition with `&& !is_support_or_admin_or_me(req.system, req.account, account)`, for less nested condition.

Reply:
Yes agree but I think it will make it more complex to read as there is another else if for `!req.system`

Reply:
I was looking at the wrong code [here](https://github.com/noobaa/noobaa-core/blob/ed421ac19606f090fc4bf1c65e07bdb1d085b003/src/server/system_services/account_server.js#L319-L321). I think for now we can keep it as same because everywhere it was like that.